public/js/projectsConstructor.js

```diff
@@ -1,49 +1,48 @@
 'use strict';
 
-var projects = [];
+var app = app || {};
 
-function Project (projectsListObj) {
-  this.title = projectsListObj.title;
-  this.url = projectsListObj.url;
-  this.githubURL = projectsListObj.githubURL;
-  this.img = projectsListObj.img;
-  this.skillsUsed = projectsListObj.skillsUsed;
-}
+var projects = [];
 
-Project.prototype.toHtml = function() {
-  var sourceHTML = $('#project-template').html();
-  var actualTemplate = Handlebars.compile(sourceHTML);
-  var newRawHTML = actualTemplate(this);
-  return newRawHTML;
-};
+(function(module) {
+  function Project (projectsListObj) {
+    this.title = projectsListObj.title;
+    this.url = projectsListObj.url;
+    this.githubURL = projectsListObj.githubURL;
+    this.img = projectsListObj.img;
+    this.skillsUsed = projectsListObj.skillsUsed;
+  }
 
-Project.loadAll = function (projectsList) {
-  projectsList.forEach(function(projectObject) {
-    projects.push(new Project(projectObject));
-  });
-};
+  Project.prototype.toHtml = function() {
+    var sourceHTML = $('#project-template').html();
+    var actualTemplate = Handlebars.compile(sourceHTML);
+    var newRawHTML = actualTemplate(this);
+    return newRawHTML;
+  };
 
-Project.initIndexPage = function() {
-  projects.forEach(function(project){
-    $('.projectsAnchor').append(project.toHtml());
-  });
-};
+  Project.loadAll = function (projectsList) {
+    projectsList.map ((projectObject) => {
+      projects.push(new Project(projectObject));
+    });
+  };
```

Comment:
`map` automatically pushes the returned item into a new array. Thus, this is not the correct way to use a `map`.

Reply:
arggghh. okay refactored back to a forEach

Reply:
You should be using a `map` here because you are creating a new array, you just need to do it slightly differently because of how a `map` works. Since we are pushing new `Project` objects into the `projects` array, you just need to rewrite it as:
```
projects = projectsList.map((projectObject) => new Project(projectObject));
```
Whatever is returned in the callback of the `map` is automatically pushed into a new array. Thus this is equivalent to your `forEach`.